lib/Sema/TypeCheckType.cpp

```diff
@@ -4481,14 +4481,17 @@ TypeResolver::resolveIsolatedTypeRepr(IsolatedTypeRepr *repr,
 
   Type type = resolveType(repr->getBase(), options);
 
+  Type unwrappedType = type;
   if (auto ty = dyn_cast<DynamicSelfType>(type)) {
-    type = ty->getSelfType();
+    unwrappedType = ty->getSelfType();
```

Comment:
I think this needs to happen after unwrapping the optional below, otherwise we will not allow `isolated Self?`.  We do generally allow optional `Self` as a special case to the covariant `Self` restrictions.

Reply:
Nice catch, fixing that